### ISSUE TITLE
Is SVM a good choice for large amount class classification?

### ISSUE THREAD

**Issue by posutsai** [davidsandberg/facenet#459]
I am trying to use SVM to classify unknown image. In my use case, I have almost 1400 classes of people to classify. But when I apply SVM on my pipeline the performance is so bad that I really doubt that if SVM is the best choice.
My pipeline is shown as below.

1. Collect images in each class
  The amount of face of each person is not fixed, from about max 600 to min 20. But the angle of face is not fixed also.

2. Embed by facenet & remove outlier
  After collecting all class image, input all of them to facenet class by class. I remove outlier in  each class using [Local Outlier Factor](http://scikit-learn.org/stable/modules/generated/sklearn.neighbors.LocalOutlierFactor.html) and set n_neighbors=10

3. Apply SVM as final classifier
  Here is the most difficult stage I meet. After survey on the internet I think maybe the reason is that output of facenet 128 dimension vector which is much less than the number of classes.

Apart from the reason I mentioned (dimension difference between embedding vector & amount of classes) I think there are still somewhere may cause such problem.

- the size of bounding box is too small or resolution may too low

- the algorithm I choose or some parameter can be tuned

I hope I describe my problem well enough for everyone to understand. If there is still some information I need to provide, please let me know. Thank you.

**Comment by mingfengwuye** [comment]
My experience is that in large amount class classification, SVM works not as well as small dataset.

**Comment by posutsai** [comment]
Which method you would recommend in this kind of situation?
I think what if I merge some similar classes into big class, make two layers of classifiers. Try to reduce amount of classes each time.
Have you ever try this on practical scenario?
If so, will this be solution of my issue?

**Comment by pavankumarkatakam** [comment]
I actually use lshforest for classification, it doesn't give accuracy rather gives the nearest face embedding distance so that i can keep another filter while recognizing the face like checking the distance with distance threshold.

**Comment by xvdehao** [comment]
@pavankumarkatakam Hi, can you give your codes which is used lshforest model?
Now I only use a SVM model and I think the accuray of face recognition could be improved.
Thanks
